src/Microsoft.OData.Core/Json/IJsonWriter.cs

```diff
@@ -175,5 +177,243 @@ public interface IJsonWriter
+        /// <param name="value">Guid value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(Guid value);
+
+        /// <summary>
+        /// Asynchronously writes a decimal value
+        /// </summary>
+        /// <param name="value">Decimal value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(decimal value);
+
+        /// <summary>
+        /// Asynchronously writes a DateTimeOffset value
+        /// </summary>
+        /// <param name="value">DateTimeOffset value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(DateTimeOffset value);
+
+        /// <summary>
+        /// Asynchronously writes a TimeSpan value
+        /// </summary>
+        /// <param name="value">TimeSpan value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(TimeSpan value);
+
+        /// <summary>
+        /// Asynchronously writes a byte value.
+        /// </summary>
+        /// <param name="value">Byte value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(byte value);
+
+        /// <summary>
+        /// Asynchronously writes an sbyte value.
+        /// </summary>
+        /// <param name="value">SByte value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(sbyte value);
+
+        /// <summary>
+        /// Asynchronously writes a string value.
+        /// </summary>
+        /// <param name="value">String value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(string value);
+
+        /// <summary>
+        /// Asynchronously writes a byte array.
+        /// </summary>
+        /// <param name="value">Byte array to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(byte[] value);
+
+        /// <summary>
+        /// Asynchronously writes a Date value
+        /// </summary>
+        /// <param name="value">Date value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(Date value);
+
+        /// <summary>
+        /// Asynchronously writes a TimeOfDay value
+        /// </summary>
+        /// <param name="value">TimeOfDay value to be written.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        Task WriteValueAsync(TimeOfDay value);
+
+#if NETCOREAPP3_1_OR_GREATER

```

Comment:
nit:

could be simplified since we're sure all .NET (core) versions we support also support this API:
```suggestion
#if NETCOREAPP
```